### DISCUSSION TITLE
Adonis multi tenancy

### DISCUSSION THREAD

**Discussion by Jon710** [adonisjs/core#3014] [General]
Has anyone implemented multi tenancy with Adonis? Are there any libs or tutorials to help me implementing it?

**Comment by anthrogan** [comment]
In most multi-tenant setups, you authorize your tokens in one place, but the scope of the tokens dictates if it's allowed to cross tenancy. We implemented this with Azure AD, it's a simple middleware that uses the jwt & jwks-rsa libraries. We control what the token has access to in the azure app registration API control module. So, in short, follow the documentation of OAuth or similar on your tenant and then give that token access to what it needs.
